feat: Be able to login to bare-metal studio easily. (#35172)
openedx/edx-platform: README.rst

  ./manage.py lms collectstatic
  ./manage.py cms collectstatic

Setup Studio SSO for Development::

  ./manage.py lms manage_user studio_worker example@example.com --unusable-password

>   # For production system do not provide the id and sceret. They should be autogenerated.

Reviewer: @feanil this comment is confusing -- do you mean that they should autogenerate them and then provide them here? or they will be set automatically by some other method?

Author: ```suggestion
  # The command below creates the application that studio uses to login.  For convenience we set the `client-id` 
  # and `client-secret` here, however in a production system you should let those be auto-generated and look 
  # them up in the django-admin so that you can set the corresponding studio settings with less guessable 
  # settings.
  #
  # The relevant studio settings are `SOCIAL_AUTH_EDX_OAUTH2_KEY` and `SOCIAL_AUTH_EDX_OAUTH2_SECRET`
```

What do you think about something like this?

Reviewer: it's still unclear to me @feanil . what autogenerates them? where are they to be found in django admin? once they are found in django admin, should this command be run with those values?

  ./manage.py lms collectstatic
  ./manage.py cms collectstatic

> 

Reviewer: commit should be `docs:` not `feat:`

Author: It's feat because of the change to devstack.py which adds new behavior being able to take a setting from the config file but happy to split them if you'd like.

Reviewer: Fair enough

> Setup Studio SSO for Development::

Reviewer: ```suggestion
Set up CMS SSO for Development::
```

two more nits-- set up as a verb is two words, and CMS is the name of the django service that includes Studio, despite the fact that we call the whole service Studio about 25% of the time

>   ./manage.py lms manage_user studio_worker example@example.com --unusable-password

Author: Split the dev and prod steps into separate sections.

> Setup Studio SSO for Development::
> 
>   ./manage.py lms manage_user studio_worker example@example.com --unusable-password
>   # DO NOT DO THIS IN PRODUCTION it will make your auth insecure
>   ./manage.py lms create_dot_application --grant-type authorization-code --skip-authorization --redirect-uris "http://localhost:18010/complete/edx-oauth2/" --scopes "user_id" studio-sso studio_worker --client-id studio-sso-key --client-secret studio-sso-secret

Reviewer: ```suggestion
Set up CMS SSO (for Development)::

  ./manage.py lms manage_user studio_worker example@example.com --unusable-password
  # DO NOT DO THIS IN PRODUCTION. It will make your auth insecure.
  ./manage.py lms create_dot_application studio-sso-id studio_worker \
      --grant-type authorization-code \
      --skip-authorization \
      --redirect-uris 'http://localhost:18010/complete/edx-oauth2/' \
      --scopes user_id  \
      --client-id 'studio-sso-id' \
      --client-secret 'studio-sso-secret'
```

* setup is two words when it's a verb
* the service is called CMS, not Studio
* use parentheses to show that it's one step with two variations (dev vs prod) rather than two steps that you're supposed to perform in series
* break up long shell commands for readability
* put positional args at the beginning for readability

> Setup Studio SSO for Production::
> 
>   ./manage.py lms manage_user studio_worker <email@yourcompany.com> --unusable-password
>   ./manage.py lms create_dot_application --grant-type authorization-code --skip-authorization --redirect-uris "http://localhost:18010/complete/edx-oauth2/" --scopes "user_id" studio-sso studio_worker
>   # Login to the django admin site (eg. http://localhost:18000/admin/oauth2_provider/application/)
>   # to view the credentials for the 'studio-sso' app and set them in the config
>   # for your production studio deployment using the `SOCIAL_AUTH_EDX_OAUTH2_KEY`
>   # and `SOCIAL_AUTH_EDX_OAUTH2_SECRET` values in your studio settings.
> 

Reviewer: ```suggestion
Set up CMS SSO (for Production):

* Create the CMS user and the OAuth application::

    ./manage.py lms manage_user studio_worker <email@yourcompany.com> --unusable-password
    ./manage.py lms create_dot_application studio-sso-id studio_worker \
        --grant-type authorization-code \
        --skip-authorization \
        --redirect-uris 'http://localhost:18010/complete/edx-oauth2/' \
        --scopes user_id

* Log into Django admin (eg. http://localhost:18000/admin/oauth2_provider/application/),
  click into the application you created above (``studio-sso-id``), and copy its "Client secret".
* In your private LMS_CFG yaml file or your private Django settings module:

   * Set ``SOCIAL_AUTH_EDX_OAUTH2_KEY`` to the client ID (``studio-sso-id``).
   * Set ``SOCIAL_AUTH_EDX_OAUTH2_SECRET`` to the client secret (which you copied).
```
* to match the dev version, why not use `studio-sso-id` instead of `studio-sso`?
* I think your code comment would be easier to follow if spelled out as bullet points outside the code block.
* change advice about ``SOCIAL_AUTH_EDX_OAUTH2_KEY``, as it might already be set for them.
* break up the long shell lines for readability
* put positional args at the beginning for readability

Set up CMS SSO (for Development)::

  ./manage.py lms manage_user studio_worker example@example.com --unusable-password
  # DO NOT DO THIS IN PRODUCTION. It will make your auth insecure.
  ./manage.py lms create_dot_application studio-sso-id studio_worker \
      --grant-type authorization-code \
      --skip-authorization \
      --redirect-uris 'http://localhost:18010/complete/edx-oauth2/' \
      --scopes user_id  \
      --client-id 'studio-sso-id' \
      --client-secret 'studio-sso-secret'

Set up CMS SSO (for Production):

* Create the CMS user and the OAuth application::

    ./manage.py lms manage_user studio_worker <email@yourcompany.com> --unusable-password
    ./manage.py lms create_dot_application studio-sso-id studio_worker \
        --grant-type authorization-code \
        --skip-authorization \
        --redirect-uris 'http://localhost:18010/complete/edx-oauth2/' \
        --scopes user_id

* Log into Django admin (eg. http://localhost:18000/admin/oauth2_provider/application/),
  click into the application you created above (``studio-sso-id``), and copy its "Client secret".
* In your private LMS_CFG yaml file or your private Django settings module:

>    * Set ``SOCIAL_AUTH_EDX_OAUTH2_KEY`` to the client ID (``studio-sso-id``).
>    * Set ``SOCIAL_AUTH_EDX_OAUTH2_SECRET`` to the client secret (which you copied).

Reviewer: ```suggestion
 * Set ``SOCIAL_AUTH_EDX_OAUTH2_KEY`` to the client ID (``studio-sso-id``).
 * Set ``SOCIAL_AUTH_EDX_OAUTH2_SECRET`` to the client secret (which you copied).
```

Reviewer: rst formatting
